src/hotspot/share/runtime/globals.hpp

```diff
@@ -573,6 +573,33 @@ const int ObjectAlignmentInBytes = 8;
   product(bool, PrintNMTStatistics, false, DIAGNOSTIC,                      \
           "Print native memory tracking summary data if it is on")          \
                                                                             \
+  product(intx, NMTRecordMemoryAllocations, 0, DIAGNOSTIC,                  \
+          "Record the specified number of memory allocations."              \
+          " 0: (the default) disables this feature."                        \
+          " max_jint (0x7FFFFFFF): keep recording until the process exits") \
+          range(0, max_jint)                                                \
+                                                                            \
+  product(intx, NMTRecordVirtualMemoryAllocations, 0, DIAGNOSTIC,           \
+          "Record the specified number of virtual memory allocations."      \
+          " 0: (the default) disables this feature."                        \
+          " max_jint (0x7FFFFFFF): keep recording until the process exits") \
+          range(0, max_jint)                                                \
```

Comment:
I don't get what these numbers are supposed to do :-). For example, what does it mean if `2` is supplied to these options?